Allow macOS users to follow the Linux Demo guide

ptgott opened the pull request:
Closes #55402

Edit the Linux Demo guide so a user can set up a local Teleport demo on a macOS workstation.

- Edit the Prerequisites to add instructions for installing the mkcert CA and starting a local Docker container.
- Tell local Docker users to skip the DNS setup step.
- Include instructions for running `teleport configure` with mkcert TLS credentials. This includes inlining the tls-certificate-setup.mdx partial, which is only used in the Linux Demo guide.
- Add other minor usability tweaks here and there so a user can follow the guide with a local container.

ptgott committed:
Allow macOS users to follow the Linux Demo guide

Closes #55402

Edit the Linux Demo guide so a user can set up a local Teleport demo on
a macOS workstation.

- Edit the Prerequisites to add instructions for installing the mkcert
  CA and starting a local Docker container.
- Tell local Docker users to skip the DNS setup step.
- Include instructions for running `teleport configure` with mkcert TLS
  credentials. This includes inlining the tls-certificate-setup.mdx
  partial, which is only used in the Linux Demo guide.
- Add other minor usability tweaks here and there so a user can follow
  the guide with a local container.

mmcallister commented on docs/pages/linux-demo.mdx:
```diff
@@ -41,6 +41,13 @@ If you want to get a feel for Teleport commands and capabilities without setting
+
 - A Linux host with only port `443` open to ingress traffic. You must be able
   to install and run software on the host. Either configure access to the host
   via SSH for the initial setup (and open an SSH port in addition port `443`)
```
```suggestion
  via SSH for the initial setup (and open an SSH port in addition to port `443`)
```

zmb3 commented on docs/pages/linux-demo.mdx:
```diff
@@ -44,38 +44,108 @@ If you want to get a feel for Teleport commands and capabilities without setting
 
 ## Step 1/4. Configure DNS
 
 Teleport uses TLS to provide secure access to its Proxy Service and Auth
```
It's a bit awkward that the first sentence under "Configure DNS" discusses TLS, not DNS.

ptgott replied:
It doesn't seem like we need this paragraph, so I've removed it.

zmb3 commented on docs/pages/linux-demo.mdx:
```diff
@@ -44,38 +44,108 @@ If you want to get a feel for Teleport commands and capabilities without setting
+  script](https://cloud.google.com/compute/docs/instances/startup-scripts), or
+  similar.
 
   For a quick demo environment you can use to follow this guide, consider
```
Should we remove the Digital Ocean callout here? It's not well-maintained and I think we might be giving the user too many options here.

zmb3 commented on docs/pages/linux-demo.mdx:
```diff
@@ -101,67 +172,163 @@ $ curl (=teleport.teleport_install_script_url=) | bash -s (=teleport.version=)
+  `teleport-tls` directory in `/etc/`, including a TLS certificate and private
+  key for Teleport.
+
+  On the container, run the following `teleport configure` command: 
```
How did `teleport` get installed in the container?

ptgott replied:
This uses the same installation command as the guide tells you to run on a Linux server: https://github.com/gravitational/teleport/pull/55751/files#diff-941e5a3006f0201e2680fcd5b589f38a4cfd9ea113a82b2af9f9729ecc090dffR163-R167

zmb3 commented on docs/pages/linux-demo.mdx:
```diff
@@ -101,67 +172,163 @@ $ curl (=teleport.teleport_install_script_url=) | bash -s (=teleport.version=)
+   Community Edition.
+
+   If you are running Teleport on a local Docker container, visit
+   https://localhost:3030.
```
Should we use 3080 to avoid confusing users with yet another port?

ptgott committed:
Respond to zmb3 feedback

- Remove unnecessary paragraph in Step 1.
- Remove DigitalOcean marketplace link, since we have not been
  maintaining the Teleport 1-Click Droplet application.
- Use port 3080 instead of 3030 for the container to avoid confusing
  users.
